Fhi.HelseId/Api/Authorization/CustomAuthenticationHandler.cs

```diff
@@ -0,0 +1,203 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Claims;
+using System.Text.Encodings.Web;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authentication.JwtBearer;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.IdentityModel.Protocols.OpenIdConnect;
+using Microsoft.IdentityModel.Tokens;
+
+
+namespace Fhi.HelseId.Api.Authorization
+{
+    /// <summary>
+    /// This is taken from package: Microsoft.AspNetCore.Authentication.JwtBearer, class JwtBearerHandler.
+    /// See the code by going into the definition, or this link: https://github.com/dotnet/aspnetcore/blob/99f2ba5fd1d668fecbff1e1a9ba6a541c99bec79/src/Security/Authentication/JwtBearer/src/JwtBearerHandler.cs#L15
```

Comment:
💭 - I don't want to be a stickler here, an @OsirisTerje probably have some thoughts on this: The original source code falls under MIT, and with MIT we need to include the original license and copyright.

https://www.tldrlegal.com/license/mit-license

Reply:
Yep, den må med.   Se https://github.com/nunit/nunit/blob/master/THIRD_PARTY_NOTICES.md  for eksempel på hvordan det kan gjøres.

Reply:
Og, må vi kopiere inn koden, kan vi ikke bruke pakken istedet?

Reply:
Koden måtte kopieres for å tillate DPOP i stede for Bearer. Altå kun ordet før jwt tokenet. Ellers er valideringen lik.

Det må til for at qt serveren selv skal bestemme om man skal aktivere dpop. Om dpop ikke aktiveres må vi støtte at klienten sener dpop token.

Reply:
Har dere sett på samplet her:  https://github.com/NorskHelsenett/HelseID.Samples/tree/master/ApiAccess ? Der er DPop implementert.

Reply:
Og her:  https://github.com/NorskHelsenett/HelseID.Samples/blob/3ff5c6ead8f3a775479880c2989b9cf4a9290e61/SampleApi/Startup.cs#L82

Reply:
Fra  @runegri  

> Rune Andreas Grimstad (Innleid)
  2:25 PM
man må det. et DPoP-token skal ikke brukes på samme måte som et Bearer token
2:25
Sjekk også sikkerhetsprofilen vår for api-er: https://utviklerportal.nhn.no/informasjonstjenester/helseid/protokoller-og-sikkerhetsprofil/sikkerhetsprofil/docs/sikkerhetsprofil_no_nbmd/
[2:25](https://norskhelsenett.slack.com/archives/D018U51B684/p1718799947659419)
Helt nederst her
2:25
API-et må tilby et endepunkt som som støtter DPoP. API-et kan tilby et separat endepunkt som bruker Bearer-tokens for å være bakoverkompatibel med gamle klienter. Et API skal ikke akseptere både Bearer-tokens og [DPoP](https://utviklerportal.nhn.no/informasjonstjenester/helseid/bruksmoenstre-og-eksempelkode/bruk-av-helseid/docs/dpop/dpop_no_nbmd/)-tokens på det samme endepunktet.

Reply:
Om du leser beskrivelsen i issuet jeg laget forklarer det problemstillingen. Jeg ser ikke hvordan det rune sier er relatert. Vi snakker ikke her om å støtte dpop, men at man skal kunne migrere på en god måte uten å måtte deploye server og klient med dpop samtidig.